### ISSUE TITLE
Methods imported from base classes defined with interface_impl do not get filtered to remove const overloads

### ISSUE THREAD

**Issue by murraypaul** [swig/swig#1277]
```%module TestMultiple

%include <swiginterface.i>
%interface_impl(Base);

class Base
{
public:
	virtual void Method();
	virtual void Method() const;
};

class Derived : public Base
{
};
```
Produces the messages from swig-3.0.12:
```
test.i(12) : Warning 516: Overloaded method Base::Method() const ignored,
test.i(11) : Warning 516: using Base::Method() instead.
```
And BaseSwigImpl.cs correctly only includes one Method().
However, Derived.cs includes two, and so fails to compile:
```
  public virtual void Method() {
    TestMultiplePINVOKE.Derived_Method__SWIG_0(swigCPtr);
  }

  public virtual void Method() {
    TestMultiplePINVOKE.Derived_Method__SWIG_1(swigCPtr);
  }
```
The workaround is to %ignore the const version of the method in the base class. Ignoring it in the derived class does not work.

**Comment by murraypaul** [comment]
This is still an issue in swig 4.0.2

**Comment by ojwb** [comment]
Also affects Java (only C# and Java have `swiginterface.i`) and still present in current git master.

**Comment by wsfulton** [comment]
With the fix in b6ece11fc1d4120a1f1ed85d1c18639fe04a34b5, you will still see the warning, both for the `Base` class and `Derived` class. However, if you `%ignore` one of the two overloaded `Method` methods in the `Base` class, the method is no longer added to the `Derived` class.
